docs/testing.md

```diff
@@ -21,16 +21,18 @@ Be sure to first `make && sudo make install`
 
 ```bash
 # Test all with nerdctl (rootless mode, if running go as a non-root user)
-go test ./cmd/nerdctl/...
+go test -p 1 ./cmd/nerdctl/...
```

Comment:
Thanks, I think this is the reason why we have failed to split `*_test.go`.

In the document could you explain what is `-p` and how it differs from `-parallel`?
The explanation should be copied to GHA yamls too.

Reply:
I also wonder if we can print a human-readable error when `-p 1` is not specified.

Reply:
Also wonder how this affects the run-time of the CI

Reply:
> Also wonder how this affects the run-time of the CI

Theoretically, this MAY impact our runtime.

Currently, we first run sequentially any test that is not parallel. Then we run all tests that can be parallelized together, with parallelization at GOMAXPROCS.

With the new model, the tests that can be parallelized are run together, but only inside the subpackage (and each subpackage is run sequentially).

But then, how many cores do we have on github actions? 4?

Back of the envelop:
- 15 subpackages
- about 1000 tests -> ~70 tests per package
- 160 calls to t.Parallel (but then, for loops, so, let's say 300) -> ~20 tests marked parallel

If we do indead have a small number of cores (<=4), then it is likely close to zero impact...

Reply:
> I also wonder if we can print a human-readable error when `-p 1` is not specified.

Maybe?

We could have the test initialization method to check the status of an flock and bail-out with an error if it is in use?
A bit heavy-handed, but then, it might work.

Reply:
> Thanks, I think this is the reason why we have failed to split `*_test.go`.
> 
> In the document could you explain what is `-p` and how it differs from `-parallel`? The explanation should be copied to GHA yamls too.

Yes.
This is not fully trivial knowledge.

Done with latest push.